### PR TITLE
firefox: pin icu to 72, fix timezone issue

### PR DESCRIPTION
**Pull request by kirillrdy**
###### Description of changes

temp workaround for https://github.com/NixOS/nixpkgs/issues/238025

after recent update to icu73, firefox fails to detect local timezone

this is caused by https://github.com/unicode-org/icu/pull/2323 interacting with firefox sandbox

I've filed a bug upstream https://bugzilla.mozilla.org/show_bug.cgi?id=1839287

meanwhile I think just pinning icu for firefox is ok, alternative fix could be wrapping firefox in TZ=:/etc/localtime

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/browsers/firefox/common.nix [hunk 1]
```diff
@@ -54,7 +54,7 @@
 , glib
 , gnum4
 , gtk3
-, icu
+, icu72 #https://bugzilla.mozilla.org/show_bug.cgi?id=1839287
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/browsers/firefox/common.nix:57]:
```suggestion
, icu72
```

File: pkgs/applications/networking/browsers/firefox/common.nix [hunk 2]
```diff
@@ -440,7 +440,7 @@ buildStdenv.mkDerivation ({
     freetype
     glib
     gtk3
-    icu
+    icu72
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/browsers/firefox/common.nix:443]:
```suggestion
    # icu73 changed how it follows symlinks which breaks in the firefox sandbox
    # https://bugzilla.mozilla.org/show_bug.cgi?id=1839287
    icu72
```

Comment (kirillrdy) [reply pkgs/applications/networking/browsers/firefox/common.nix:445]: yes, this is much better

**Comment by mweinelt** [comment]
Updated the commit message to use the correct attribute and expanded on why this issue happens.

Unfortunately I could not reproduce this locally, but I'm not opposed to pinning to icu72 if it helps users and the issue gets tracked upstream.

**Comment by kirillrdy** [comment]
@mweinelt thank you,

two questions
1. should we backport this to 23.05 ?
2. since you are not the first who cant reproduce the issue, is it hard for you provide to build-vm setup that I can have a look at ? ( curiosity )

**Comment by mweinelt** [comment]
Backport in https://github.com/NixOS/nixpkgs/pull/238853.

**Pull request merged**